include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -243,6 +244,32 @@ inline ConstMemoryInfo AllocatorImpl<T>::GetInfo() const {
   return ConstMemoryInfo{out};
 }
 
+template <typename T>
+inline std::unordered_map<std::string, std::string> AllocatorImpl<T>::GetStats() const {
+  AllocatorWithDefaultOptions allocator;
+  char* raw_stats = nullptr;
+  ThrowOnError(GetApi().AllocatorGetStats(this->p_, allocator, &raw_stats));
+  std::unordered_map<std::string, std::string> stats;
+  if (raw_stats == nullptr) {
+    return stats;
+  }
+
+  auto free_fn = detail::AllocatedFree(allocator);
+  std::unique_ptr<void, decltype(free_fn)> raw_stats_g(raw_stats, free_fn);
+
+  std::istringstream iss(raw_stats);
+  std::string line;
+  while (std::getline(iss, line, ',')) {
```

Comment:
> ',')) { [](http://example.com/codeflow?start=32&length=8)

My understanding that lines are no longer separated by comma. Is your test detecting this code issue?

Reply:
I think command-separated string is more formal as API response. Thus, I postproc the debug string in allocator_adapter to replace newline with comma and remove redundant spaces.